processor/src/operations/horner_ops.rs

```diff
@@ -0,0 +1,515 @@
+    ///    of 4 extension field coefficients of the polynomial.
+    /// 2. (acc0, acc1) stands for an extension field element accumulating the values of the Horner
+    ///    evaluation procedure. (acc0', acc1') is the updated value of this accumulator.
+    /// 3. t_addr is the memory address from which we are loading the ci's when using the MSTREAM
+    ///    instruction or storing using the ADVPIPE instruction.
+    /// 4. r_addr is the memory address of the evaluation point i.e., alpha.
+    ///
+    /// The instruction also makes use of the helper registers to hold the value of
+    /// alpha = (alpha0, alpha1) during the course of its execution.
+    pub(super) fn op_horner_eval_ext(&mut self) -> Result<(), ExecutionError> {
+        // --- read the values of the coefficients, over the extension field, from the stack ------
+        let coef = self.get_coeff_as_quad_ext_elements();
+
+        // --- read the randomness from memory ----------------------------------------------------
+        let alpha = self.get_random_point()?;
+
+        // --- read the accumulator values from stack ---------------------------------------------
+        let acc_old = self.get_accumulator();
+
+        // --- compute the updated accumulator values ---------------------------------------------
+        let acc_new = coef.iter().rev().fold(acc_old, |acc, coef| *coef + alpha * acc);
+
+        // --- copy the stack ---------------------------------------------------------
+        self.stack.copy_state(0);
+
+        // --- update the accumulators ------------------------------------------------------------
+        self.stack.set(14, acc_new.to_base_elements()[1]);
+        self.stack.set(15, acc_new.to_base_elements()[0]);
+
+        // --- set the helper registers -----------------------------------------------------------
+        self.populate_helper_registers(alpha);
+
+        Ok(())
+    }
+
+    //// HELPER METHODS
+    //// ------------------------------------------------------------------------------------------
+
+    /// Returns the top 8 elements of the operand stack.
+    fn get_coeff_as_base_elements(&self) -> [Felt; 8] {
+        let c0 = self.stack.get(0);
+        let c1 = self.stack.get(1);
+        let c2 = self.stack.get(2);
+        let c3 = self.stack.get(3);
+        let c4 = self.stack.get(4);
+        let c5 = self.stack.get(5);
+        let c6 = self.stack.get(6);
+        let c7 = self.stack.get(7);
+
+        [c0, c1, c2, c3, c4, c5, c6, c7]
+    }
+
+    /// Returns the top 8 elements of the operand stack.
+    fn get_coeff_as_quad_ext_elements(&self) -> [QuadFelt; 4] {
+        let c0 = self.stack.get(0);
+        let c1 = self.stack.get(1);
+        let c2 = self.stack.get(2);
+        let c3 = self.stack.get(3);
+        let c4 = self.stack.get(4);
+        let c5 = self.stack.get(5);
+        let c6 = self.stack.get(6);
+        let c7 = self.stack.get(7);
```

Comment:
minor nit: could we name those variables like in the docstring, i.e. `c0_1`, `c0_0`, etc?

Reply:
Changed